test/test_images/sendeventswithtracing/main.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	gohttp "net/http"
+	"os"
+	"strconv"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/plugin/ochttp/propagation/b3"
+	"knative.dev/pkg/tracing"
+)
+
+var (
```

Comment:
FWIW, I tend to prefer using env vars, rather than flags these days. It would be fairly easy to turn this into a struct with envconfig tags. If you think that it is more important to be easy to run manually, then feel free to leave it as flags.